sympy/solvers/solveset.py

```diff
@@ -1367,23 +1365,21 @@ def _solve_as_lambert(lhs, rhs, symbol, domain):
 
 def _solve_as_lambert_bivariate(lhs, rhs, symbol, domain):
 
-    result = ConditionSet(symbol, Eq(lhs-rhs, 0), domain)
+    result = ConditionSet(symbol, Eq(lhs - rhs, 0), domain)
 
     soln = _solve_as_lambert(lhs, rhs, symbol, domain)
-    if soln:
-        result = FiniteSet(*soln)
-    else:
+    if not soln:
         # make symbol positive
         u = Dummy('u', positive=True)
         pos_lhs = lhs.subs({symbol: u})
-        soln = _solve_as_lambert(pos_lhs, rhs, u, domain)
-        if soln:
-            result = FiniteSet(*soln)
-        else:
-            # maybe bivariate type
+        dom = Intersection(domain, Interval(0, oo))
```

Comment:
I guess modifying the domain rather than using  "ConditionSet" will be better as the `symbol` to be solved needs to be positive to get the solutions.

Reply:
Is there a solution being missed?